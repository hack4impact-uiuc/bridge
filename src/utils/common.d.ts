declare module "@hack4impact-uiuc/bridge/common" {
  import { Props, ReactType } from "react";
  import {
    ColorProps,
    SpaceProps,
    LayoutProps,
    TypographyProps as StyledTypographyProps,
    BordersProps as StyledBordersProps,
    BoxShadowProps,
  } from "styled-system";

  export interface BaseProps extends Props<any> {
    as?: ReactType;
    className?: string;
  }
  export interface CommonProps
    extends BaseProps,
      ColorProps,
      SpaceProps,
      LayoutProps {}

  export interface TypographyProps extends BaseProps, StyledTypographyProps {}

  export interface BorderProps extends StyledBordersProps, BoxShadowProps {}
}
