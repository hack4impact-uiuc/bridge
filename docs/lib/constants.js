export const COMPONENTS_URL = '/components';
export const DESIGN_URL = '/design';
