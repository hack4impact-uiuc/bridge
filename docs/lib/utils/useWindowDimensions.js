import { useState, useEffect } from 'react';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useWindowDimensions() {
  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());

  useEffect(() => {
    function handleResize() {
      const { innerWidth: width, innerHeight: height } = window;
      setWindowDimensions({ width, height });
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowDimensions;
}
