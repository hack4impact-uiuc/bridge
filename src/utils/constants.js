import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
