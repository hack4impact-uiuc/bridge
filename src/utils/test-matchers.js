/* eslint-disable react/forbid-foreign-prop-types */
// taken from https://github.com/primer/components/blob/master/src/utils/test-matchers.js
import 'jest-styled-components';

const stringify = (d) => JSON.stringify(d, null, '  ');
/**
 * These are props that styled-system aliases for backwards compatibility.
 * For some reason, they don't show up in our toImplementSystemProps() matcher,
 * so we skip over them.
 */
const ALIAS_PROP_TYPES = ['w', 'align', 'justify', 'wrap'];

// eslint-disable-next-line no-undef
expect.extend({
  toImplementSystemProps(Component, propNames) {
    const propKeys = new Set(Object.keys(Component.propTypes));
    const expectedPropKeys = Object.keys(propNames.propTypes);
    const missing = expectedPropKeys.filter((key) => !propKeys.has(key))
      .filter((key) => !ALIAS_PROP_TYPES.includes(key));
    return {
      pass: missing.length === 0,
      message: () => `Missing prop${missing.length === 1 ? '' : 's'}: ${stringify(missing)}`,
    };
  },
});
