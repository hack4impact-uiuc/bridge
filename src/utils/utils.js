import themeGet from '@styled-system/theme-get';
import * as styledSystem from 'styled-system';
import theme from '../theme';

const { get: getKey } = styledSystem;

export const get = (key) => themeGet(key, getKey(theme, key));

export const getOrDefault = (val, def) => (val !== undefined && val !== null ? val : def);

export const lodashGet = (object, path, defaultVal) => {
  const PATH = Array.isArray(path)
    ? path
    : path.split('.').filter((i) => i.length);
  if (!PATH.length) {
    return object === undefined ? defaultVal : object;
  }
  if (object === null || object === undefined || typeof (object[PATH[0]]) === 'undefined') {
    return defaultVal;
  }
  return lodashGet(object[PATH.shift()], PATH, defaultVal);
};
