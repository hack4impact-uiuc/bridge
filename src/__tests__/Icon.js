import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Icon } from '..';
import { COMMON } from '../utils/constants';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('Icon', () => {
  it('renders a <svg>', () => {
    expect(renderJSON(<Icon />).type).toEqual('svg');
  });

  it('should have no axe violations', async () => {
    const { container } = render(<Icon />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('default to 20px width', () => {
    const { container } = render(<Icon />);
    expect(container.firstChild).toHaveStyle('width: 20px;');
  });

  it('implements system props', () => {
    expect(Icon).toImplementSystemProps(COMMON);
  });

  it('respects "height" props', () => {
    const { container } = render(<Icon height={200} />);
    expect(container.firstChild).toHaveStyle('height: 200px;');
  });

  it('respects "width" props', () => {
    const { container } = render(<Icon width={200} />);
    expect(container.firstChild).toHaveStyle('width: 200px;');
  });

  it('respects "fill" props', () => {
    const { container } = render(<Icon fill="#155DA1" />);
    expect(container.firstChild).toHaveStyle('fill: #155DA1');
  });
});

// Icons match
describe('Icons types match', () => {
  it('renders errorAlert', () => {
    expect(renderJSON(<Icon type="errorAlert" />)).toMatchSnapshot();
  });
  it('renders infoAlert', () => {
    expect(renderJSON(<Icon type="infoAlert" />)).toMatchSnapshot();
  });
  it('renders successAlert', () => {
    expect(renderJSON(<Icon type="successAlert" />)).toMatchSnapshot();
  });

  it('renders warningAlert', () => {
    expect(renderJSON(<Icon type="warningAlert" />)).toMatchSnapshot();
  });

  it('renders fileDragDrop', () => {
    expect(renderJSON(<Icon type="fileDragDrop" />)).toMatchSnapshot();
  });

  it('renders formErrorAlert', () => {
    expect(renderJSON(<Icon type="formErrorAlert" />)).toMatchSnapshot();
  });

  it('renders opacityUploadFile', () => {
    expect(renderJSON(<Icon type="opacityUploadFile" />)).toMatchSnapshot();
  });

  it('renders uploadFile', () => {
    expect(renderJSON(<Icon type="uploadFile" />)).toMatchSnapshot();
  });
});
