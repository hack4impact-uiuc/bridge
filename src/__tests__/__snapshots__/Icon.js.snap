// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons types match renders errorAlert 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 30 30"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M15 29C22.732 29 29 22.732 29 15C29 7.26801 22.732 1 15 1C7.26801 1 1 7.26801 1 15C1 22.732 7.26801 29 15 29Z"
    fillRule="evenodd"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M19.1998 10.8L10.7998 19.2"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M10.7998 10.8L19.1998 19.2"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons types match renders fileDragDrop 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 38 47"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M23.5 1H5.5C3.01472 1 1 3.01472 1 5.5V41.5C1 43.9853 3.01472 46 5.5 46H32.5C34.9853 46 37 43.9853 37 41.5V14.5L23.5 1Z"
    fillRule="evenodd"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M24 1V14H37"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M20 37V24"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M13 31H26"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons types match renders formErrorAlert 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 18 18"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M9 17C13.4183 17 17 13.4183 17 9C17 4.58172 13.4183 1 9 1C4.58172 1 1 4.58172 1 9C1 13.4183 4.58172 17 9 17Z"
    fillRule="evenodd"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="1.25"
  />
  <path
    d="M8.99961 5.7998V8.9998"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="1.25"
  />
  <path
    d="M8.99952 12.1998H9.00752"
    stroke="#A62B31"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="1.25"
  />
</svg>
`;

exports[`Icons types match renders infoAlert 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 30 30"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M15 29C22.732 29 29 22.732 29 15C29 7.26801 22.732 1 15 1C7.26801 1 1 7.26801 1 15C1 22.732 7.26801 29 15 29Z"
    fillRule="evenodd"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M14.9998 20.6V15"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M15.0001 9.39995H15.0141"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons types match renders opacityUploadFile 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 22 27"
  width="20px"
>
  <g
    opacity="0.5"
  >
    <path
      clipRule="evenodd"
      d="M13.5 1H3.5C2.11929 1 1 2.08394 1 3.42105V22.7895C1 24.1266 2.11929 25.2105 3.5 25.2105H18.5C19.8807 25.2105 21 24.1266 21 22.7895V8.26316L13.5 1Z"
      fillRule="evenodd"
      stroke="#155DA1"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M13.6318 1V8.36842H21.0003"
      stroke="#155DA1"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M16.789 14.1582H6.2627"
      stroke="#155DA1"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M16.789 19.4208H6.2627"
      stroke="#155DA1"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M8.36883 9.94721H6.78988H5.21094"
      stroke="#155DA1"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </g>
</svg>
`;

exports[`Icons types match renders successAlert 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 30 31"
  width="20px"
>
  <path
    d="M29.0001 14.1121V15.4001C28.9965 21.6002 24.9153 27.0596 18.9696 28.8175C13.0239 30.5755 6.63003 28.2133 3.25542 23.012C-0.119198 17.8106 0.329288 11.0092 4.35766 6.29599C8.38604 1.58279 15.0347 0.0806215 20.6981 2.60409"
    stroke="#256830"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M28.9998 4.19995L14.9998 18.214L10.7998 14.014"
    stroke="#256830"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons types match renders uploadFile 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 22 27"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M13.5 1H3.5C2.11929 1 1 2.08394 1 3.42105V22.7895C1 24.1266 2.11929 25.2105 3.5 25.2105H18.5C19.8807 25.2105 21 24.1266 21 22.7895V8.26316L13.5 1Z"
    fillRule="evenodd"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M13.6318 1V8.36842H21.0003"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M16.789 14.1582H6.2627"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M16.789 19.4208H6.2627"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M8.36883 9.94721H6.78988H5.21094"
    stroke="#155DA1"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons types match renders warningAlert 1`] = `
.c0 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

<svg
  className="c0"
  fill="none"
  viewBox="0 0 30 26"
  width="20px"
>
  <path
    clipRule="evenodd"
    d="M12.8238 2.36716L2.04375 20.3635C1.59148 21.1468 1.58878 22.1112 2.03666 22.8969C2.48454 23.6827 3.31573 24.1718 4.22012 24.1817H25.7801C26.6845 24.1718 27.5157 23.6827 27.9636 22.8969C28.4115 22.1112 28.4088 21.1468 27.9565 20.3635L17.1765 2.36716C16.715 1.60642 15.8899 1.14185 15.0001 1.14185C14.1104 1.14185 13.2852 1.60642 12.8238 2.36716Z"
    fillRule="evenodd"
    stroke="#FFD15F"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M15.0001 8.90918V14.0001"
    stroke="#FFD15F"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M15.0001 19.091H15.0128"
    stroke="#FFD15F"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;
